### PR TITLE
Implement Navigation/Scroll Functionality

### PR DESCRIPTION
**Pull request by samuelsogbesan**
Add Navigation Functionality

- Connect click event of each navigation card to a navigate(), a function that smoothly scrolls to the given section.
- This connection is achieved through the data-link custom attribute on each navigation element.
- The helper initialiseNavigation is used to hook up the cards onload.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -10,4 +10,15 @@
...
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
+const navigate = (pageid) => document.getElementById(pageid).scrollIntoView({behavior:"smooth",block:"start"});
+
+const initialiseNavigation = () => {
+  var navigationItems = document.getElementsByClassName("navigation-item");
+  for(var i = 0; i <navigationItems.length;i++) navigationItems[i].addEventListener("click",function(){
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: [add a space between `function()` and `{`](https://google.github.io/styleguide/jsguide.html#formatting-nonempty-blocks)

Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: add a space between `;` and `i++`

Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: add a space between `for` and `(`

Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: [prefer a `for - of` loop, if possible](https://google.github.io/styleguide/jsguide.html#features-for-loops)

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -10,4 +10,15 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
+const navigate = (pageid) => document.getElementById(pageid).scrollIntoView({behavior:"smooth",block:"start"});
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: [use single quotes](https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes), here and elsewhere

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: [use braces for control statements](https://google.github.io/styleguide/jsguide.html#formatting-braces-all)

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: [pick one of the valid formatting styles for object literals](https://google.github.io/styleguide/jsguide.html#formatting-object-literals)

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -11,10 +11,11 @@ body {
 }
 
 #fullpage {
-  min-width: 100vw;
+min-width: 100vw;
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/style.css]: indent and maintain sorted order

File: portfolio/src/main/webapp/style.css [hunk 4]
```diff
@@ -11,10 +11,11 @@ body {
...
   min-height: 100vh;
   overflow-x: hidden;
   overflow-y: scroll;
+  scroll-behavior:smooth;
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/style.css]: add a space after `:`

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -10,4 +10,15 @@
...
+
+const initialiseNavigation = () => {
+  var navigationItems = document.getElementsByClassName("navigation-item");
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: [don't use `var`](https://google.github.io/styleguide/jsguide.html#features-use-const-and-let)

**Review by krasimirgg — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: changed to for - of here 6bff026

Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Added in this commit with loop type change 6bff026

Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Space added here a44099c

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Updated with Commits 6926198 and 3f1c720

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Updated in commit e22a785

File: portfolio/src/main/webapp/style.css (hunk 3 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/style.css]: Indented here 17b27e0 and Alphabetised here e27539f

File: portfolio/src/main/webapp/style.css (hunk 4 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/style.css]: Space added with commit 80fda63

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Changed from var to const with commit 7a83ce2

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Format updated with commit d091e22
Note: I'm a little unsure here. Referring to [4.2.2](https://google.github.io/styleguide/jsguide.html#formatting-object-literals), I am attempting to style my object literal as described in the second example. Is the missing part the space between the ',' and the 'block' attribute?

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: For loop type has been changed here 6bff026

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -10,4 +10,16 @@
...
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
+const navigate = (pageid) => document.getElementById(pageid).scrollIntoView({behavior:'smooth', block:'start'});
+
+const initialiseNavigation = () => {
+  const navigationItems = document.getElementsByClassName('navigation-item');
+  for (const item of navigationItems) {item.addEventListener('click',function() {
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: add a line break between `{` and `item`

Comment (krasimirgg) [reply portfolio/src/main/webapp/script.js]: add a space between `,` and `function`

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -10,4 +10,16 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
+const navigate = (pageid) => document.getElementById(pageid).scrollIntoView({behavior:'smooth', block:'start'});
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: [object literals](https://google.github.io/styleguide/jsguide.html#formatting-object-literals) have a space between the `key:` and `value`, as in `key: 'value'`.

Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: rename `pageid` to `pageId`, using [`lowerCamelCase`](https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined)

**Review by krasimirgg — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Updated in commit 9e98000

Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Rectified in commit 49d60b3

Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Updated in commit 9e98000

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Updated in commit 020389c

**Review by krasimirgg — APPROVED** [review]
Nice!

**Pull request merged**